components/openvsx-proxy/pkg/modifyresponse.go

```diff
@@ -48,19 +50,53 @@ func (o *OpenVSXProxy) ModifyResponse(r *http.Response) error {
 		return nil
 	}
 
-	body, err := ioutil.ReadAll(r.Body)
+	rawBody, err := ioutil.ReadAll(r.Body)
 	if err != nil {
-		log.WithFields(logFields).WithError(err).Error("error reading response body")
+		log.WithFields(logFields).WithError(err).Error("error reading response raw body")
 		return err
 	}
 	r.Body.Close()
-	r.Body = ioutil.NopCloser(bytes.NewBuffer(body))
+	r.Body = ioutil.NopCloser(bytes.NewBuffer(rawBody))
+
+	if r.StatusCode >= 300 && r.StatusCode < 400 && o.Config.FollowRedirects > 0 && r.Header.Get("Location") != "" {
```

Comment:
@jeanp413 are these changes necessary? The problem was that open-vsx.org has a response with URLs to open-vsx.org and since compressions is enabled we did not replace them?

Why do we need to tackle redirects?

Reply:
All workspaces are configured to use  `http_proxy` and `no_proxy` in their setup. It is fine. We don't fix this issue for air-gapped but for general case:
- open-vsx.org is not reliable, request to it should go through OpenVSX proxy
- requests to openvsxorg.blob.core.windows.net (and also big), so we don't cache them and go directly

The cache is only 200Mb, we don't want to populate it with something like 50Mb vsix for java extension, but fast responses for common questions to open-vsx.org.

Reply:
Thanks for the explanation, I'll remove the redirect logic then